external_parser/parse_example_binary.cc

```diff
@@ -343,41 +299,47 @@ bool binary_parser::parse_examples(vw *all, v_array<example *> &examples) {
   }
 
   unsigned int payload_type;
-
-  if (!advance_to_next_payload_type(all->example_parser->input.get(),
-                                    payload_type)) {
-    return false;
-  }
-
-  if (payload_type == MSG_TYPE_CHECKPOINT) {
-    if (!read_checkpoint_msg(all->example_parser->input.get())) {
-      return false;
-    }
-
-    if (!advance_to_next_payload_type(all->example_parser->input.get(),
-                                      payload_type)) {
-      return false;
-    }
-  }
-
-  while (payload_type == MSG_TYPE_REGULAR) {
-    if (read_regular_msg(all->example_parser->input.get(), examples)) {
-      return true;
-    }
-
-    // bad payload process the next one
-    if (!advance_to_next_payload_type(all->example_parser->input.get(),
-                                      payload_type)) {
+  while(true) {
```

Comment:
can this be `while(advance_to_next_payload_type())`?

Reply:
it can